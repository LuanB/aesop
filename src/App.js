import React, {useState, useEffect} from 'react';

import Accordion from './components/Accordion';
import Card from './components/Card';
import ProductGroup from './components/product-group/product-group.component'
import './App.css';

function App() {
  const [hasError, setErrors] = useState(false);
  const [products, setProducts] = useState(null);
  const [activeEventKey, setActiveEventKey] = useState(0);

  async function fetchData() {
    const res = await fetch("/au/api/v1/nav/shop");
    res
      .json()
      .then(res => setProducts(res))
      .catch(err => setErrors(true));
     
  }

  useEffect(() => {
    fetchData();
  }, []);

  const ErrorComponent = () => {
    return (
      <div>handle Error occurred screen</div>
    )
  }

   const AccordionComponent = () => {
    const {categories} = products;
    
    console.log(products);
    return(
      hasError ? <ErrorComponent /> :

    <Accordion>
    {categories.map(({ name, items }, index) => (
      <Card key={index}>
        <Accordion.Toggle element={Card.Header} eventKey={index}>
          {index + 1}. {name}
        </Accordion.Toggle>
        <Accordion.Collapse eventKey={index} element={Card.Body}>
         <ProductGroup items={items} />

        </Accordion.Collapse>
      </Card>
    ))}
  </Accordion>
    
  )
    }

  return (
    <div className="App">
      <div>Aesop Accordion Coding Challenge </div>
    {products ? <AccordionComponent /> : null}
    </div>
  );
}

export default App;
