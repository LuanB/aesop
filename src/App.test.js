import React from 'react';
import  {shallow, mount} from 'enzyme';
import App from './App';
import Accordion from './components/Accordion'
import Card from './components/Card'
import ProductGroup from './components/product-group/product-group.component'
import { ExpansionPanelActions } from '@material-ui/core';







describe('App container', () => {

 it("should render the components", () => {
   
  const wrapper = shallow(<App/>);

  expect(wrapper.find(<Accordion />)).toBeTruthy();
  expect(wrapper.find(<Card />)).toBeTruthy();


 })


 it("should match snapshot", () => {
   
  const wrapper = shallow(<App/>);

 expect(wrapper).toMatchSnapshot();


 })

 it("should not display Accordion if no products", () => {

    const mockData = {
        categories: [
            {name: 'Cat1', items: [{name:'Item1', items:[{name: 'Product1', price: 'From $100', thumbnail: '/test/test.png'}]}]},
            {name: 'Cat2', items: [{name:'Item2', items:[{name: 'Product2', price: 'From $200', thumbnail: '/test/test2.png'}]}]},
        ]
    }
   
    const wrapper = mount(<App/>);

    
    wrapper.setState({products: mockData, hasError: false });
    console.log(wrapper.debug());
   //expect(wrapper).containsMatchingElement(<Accordion/>).toBeTruthy();
  
  
   })

})