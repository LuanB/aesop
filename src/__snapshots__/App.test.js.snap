// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container should match snapshot 1`] = `
<div
  className="App"
>
  <div>
    Aesop Accordion Coding Challenge 
  </div>
</div>
`;
